src/apprt/gtk/App.zig

```diff
@@ -165,6 +129,93 @@ pub fn init(core_app: *CoreApp, opts: Options) !App {
+    };
+
+    var gdk_disable: struct {
+        @"gles-api": bool = false,
+        vulkan: bool = false,
+    } = .{};
+
+    environment: {
+        if (version.atLeast(4, 16, 0)) {
+            // From gtk 4.16, GDK_DEBUG is split into GDK_DEBUG and GDK_DISABLE.
+            // For the remainder of "why" see the 4.14 comment below.
+            gdk_disable.@"gles-api" = true;
+            gdk_disable.vulkan = true;
+            // gdk_debug.opengl = true;
+            gdk_debug.@"gl-no-fractional" = true;
+            break :environment;
+        }
+        if (version.atLeast(4, 14, 0)) {
+            // We need to export GDK_DEBUG to run on Wayland after GTK 4.14.
+            // Older versions of GTK do not support these values so it is safe
+            // to always set this. Forwards versions are uncertain so we'll have
+            // to reassess...
+            //
+            // Upstream issue: https://gitlab.gnome.org/GNOME/gtk/-/issues/6589
+            // gdk_debug.opengl = true;
+            gdk_debug.@"gl-disable-gles" = true;
+            gdk_debug.@"gl-no-fractional" = true;
+            gdk_debug.@"vulkan-disable" = true;
+            break :environment;
+        }
+        // Versions prior to 4.14 are a bit of an unknown for Ghostty. It
+        // is an environment that isn't tested well and we don't have a
+        // good understanding of what we may need to do.
+        gdk_debug.@"vulkan-disable" = true;
+    }
+
+    {
+        var buf: [128]u8 = undefined;
+        var fmt = std.io.fixedBufferStream(&buf);
+        const writer = fmt.writer();
+        var first: bool = true;
+        inline for (@typeInfo(@TypeOf(gdk_debug)).Struct.fields) |field| {
+            if (@field(gdk_debug, field.name)) {
+                if (!first) try writer.writeAll(",");
+                try writer.writeAll(field.name);
+                first = false;
+            }
+        }
+        try writer.writeByte(0);
+        log.warn("setting GDK_DEBUG={s}", .{fmt.getWritten()});
+        _ = internal_os.setenv("GDK_DEBUG", buf[0 .. fmt.pos - 1 :0]);
+    }
+
+    {
+        var buf: [128]u8 = undefined;
+        var fmt = std.io.fixedBufferStream(&buf);
+        const writer = fmt.writer();
+        var first: bool = true;
+        inline for (@typeInfo(@TypeOf(gdk_disable)).Struct.fields) |field| {
+            if (@field(gdk_disable, field.name)) {
+                if (!first) try writer.writeAll(",");
+                try writer.writeAll(field.name);
+                first = false;
+            }
+        }
+        try writer.writeByte(0);
+        log.warn("setting GDK_DISABLE={s}", .{fmt.getWritten()});
+        _ = internal_os.setenv("GDK_DISABLE", buf[0 .. fmt.pos - 1 :0]);
+    }
+
+    if (version.atLeast(4, 14, 0)) {
+        // We need to export GSK_RENDERER to opengl because GTK uses ngl by
+        // default after 4.14
+        _ = internal_os.setenv("GSK_RENDERER", "opengl");
+    }
```

Comment:
it would be nice to override this from the config because personally, I just want to use whatever Fedora ships by default.

Reply:
I went with `default` and `opengl`.